src/imitation/algorithms/preference_comparisons.py

```diff
@@ -906,6 +908,201 @@ def _reward_sums(self, fragment_pairs) -> Tuple[np.ndarray, np.ndarray]:
         return np.array(rews1, dtype=np.float32), np.array(rews2, dtype=np.float32)
 
 
+class SynchronousCLIGatherer(PreferenceGatherer):
```

Comment:
It occurs to me that this isn't very testable right now

Reply:
I'm going to leave this untested for now, since I have other tasks I need to attend to for now. 🤞 that I'll get to it later but I'm not sure I will